src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -718,6 +718,11 @@ static StoragePtr create(const StorageFactory::Arguments & args)
         ++arg_num;
     }
 
+    DataTypes data_types = metadata.partition_key.data_types;
+    for (size_t i = 0; i < data_types.size(); ++i)
+        if (strncmp(data_types[i]->getFamilyName(), "Float", 5) == 0)
+            throw Exception("Donot support float point as partition key: " + metadata.partition_key.column_names[i], ErrorCodes::BAD_ARGUMENTS);
```

Comment:
If should be allowed to ATTACH table even with bad partition key
(so the server can startup after version update).

See args.attach.